### PR TITLE
Add Explicit Gamelist Type selection to GUI menu.

### PR DESCRIPTION
**Pull request by zigurana**
Currently supports Basic, Detailed, Video, and Automatic types. The Automatic type checks for the availability of video's first, then screenshots, finally defaulting to Basic view if none are present.

### CODE REVIEW

**Comment by robertybob** [comment]
Nice! Whose brilliant idea was this? :D

File: es-app/src/views/ViewController.h [hunk 1]
```diff
@@ -59,7 +68,7 @@ class ViewController : public GuiComponent
 		friend ViewController;
 		SystemData* system;
 	};
-
+
```
Comment (joolswills) [inline es-app/src/views/ViewController.h]: Please check diffs for whitespace stuff like this before doing the PR.

**Comment by joolswills** [comment]
If anyone has some time to test this, it would be appreciated. Thanks.

**Comment by jrassa** [comment]
@joolswills I tested this out and its working well.

@zigurana looks like you might need to rebase.  There were some duplicate changes between this branch and your carousel branch which are now causing merge conflicts.  Also, I would recommend adjusting the name of the menu option to "GAMELIST VIEW STYLE" (adding a space between gamelist and view).

**Comment by zigurana** [comment]
@jrassa Thanks for having a look at this. I've followed your recommendation and rebased with the Retropie Master.
@joolswills : I've tried to find more whitespace issues, but I think (and hope) I have not introduced any more of them.

**Comment by joolswills** [comment]
Thanks. Does anyone have time to help me with testing of this (and other new features) ?

**Comment by Darknior** [comment]
Same, it is a great feature 👍 
If i can help to try it, tell it to me @joolswills 😃

File: es-app/src/views/ViewController.cpp [hunk 2]
```diff
@@ -231,33 +231,51 @@ std::shared_ptr<IGameListView> ViewController::getGameListView(SystemData* syste
...
+	{
+		case VIDEO:
+			view = std::shared_ptr<IGameListView>(new VideoGameListView(mWindow, system->getRootFolder()));
+			break;
+		case DETAILED:
+			view = std::shared_ptr<IGameListView>(new DetailedGameListView(mWindow, system->getRootFolder()));
+			break;
+		// case GRID placeholder for future implementation.
+		//		view = std::shared_ptr<IGameListView>(new GridGameListView(mWindow, system->getRootFolder()));
+		//		break;
```
Comment (joolswills) [inline es-app/src/views/ViewController.cpp]: please remove this - I don't think we need any reminder here for code that doesn't exist in this fork. thanks.

File: es-app/src/views/ViewController.h [hunk 3]
```diff
@@ -49,6 +49,15 @@ class ViewController : public GuiComponent
 		GAME_LIST
 	};
 
+	enum GameListViewType
+	{
+		AUTOMATIC,
+		BASIC,
+		DETAILED,
+		VIDEO
+		// GRID TODO!
```
Comment (joolswills) [inline es-app/src/views/ViewController.h]: as above.

**Comment by joolswills** [comment]
couple of minor changes - but tested and works well for me. Please remove references to the grid view and I will merge. thanks.

**Comment by joolswills** [comment]
Ill merge as-is since no-one seems to be around :)

**Pull request merged**